### ISSUE TITLE
Issue with 4.3.0 install  - seeking CUDA when none is installed?

### ISSUE THREAD

**Issue by jafrank88** [BBC-Esq/VectorDB-Plugin-for-LM-Studio#156]
![image](https://github.com/BBC-Esq/VectorDB-Plugin-for-LM-Studio/assets/91024347/6d779901-4a0f-4872-940b-6b9cd55ccae9)
No CUDA and responded to setup question with 'ok to use CPU only'

**Comment by BBC-Esq** [comment]
Any updates on this before I start looking into it?

**Comment by jafrank88** [comment]
I will try again this hour and report back,

**Comment by BBC-Esq** [comment]
Can you please try the newest version and let me know if it's still and issue?

**Comment by jafrank88** [comment]
Sure. On Monday should work
Jonathan

**Comment by jafrank88** [comment]
No CUDA/Nvidia on this machine. Cheers, Jonathan

[image: image.png]

**Comment by jafrank88** [comment]
Not sure if this helps,
[image: image.png]

**Comment by BBC-Esq** [comment]
Weird, the image isn't showing for some reason...

**Comment by jafrank88** [comment]
[image: unnamed.png]
[image: image.png]

**Comment by BBC-Esq** [comment]
Can you please doublecheck whether this modification to ```setup.py``` works?  You will need to replace the "install_pytorch" function verbatim with the following code, save the file, and they try installing again.  If you get back to me today I'll probably be able to include this in the new release scheduled for today, otherwise, you'll have to do the same thing for the next release:

```
def install_pytorch(cuda_version_num, cuda_installed):
    major, minor = map(int, sys.version.split()[0].split('.')[:2])
    if cuda_installed and cuda_version_num is not None and cuda_version_num >= 12.1:
        # Install PyTorch with CUDA support
        if minor == 11:
            os.system("pip3 install https://download.pytorch.org/whl/cu121/torch-2.2.2%2Bcu121-cp311-cp311-win_amd64.whl https://download.pytorch.org/whl/cu121/torchvision-0.17.2%2Bcu121-cp311-cp311-win_amd64.whl https://download.pytorch.org/whl/cu121/torchaudio-2.2.2%2Bcu121-cp311-cp311-win_amd64.whl https://github.com/jakaline-dev/Triton_win/releases/download/3.0.0/triton-3.0.0-cp311-cp311-win_amd64.whl")
        elif minor == 10:
            os.system("pip3 install https://download.pytorch.org/whl/cu121/torch-2.2.2%2Bcu121-cp310-cp310-win_amd64.whl https://download.pytorch.org/whl/cu121/torchvision-0.17.2%2Bcu121-cp310-cp310-win_amd64.whl https://download.pytorch.org/whl/cu121/torchaudio-2.2.2%2Bcu121-cp310-cp310-win_amd64.whl https://github.com/jakaline-dev/Triton_win/releases/download/3.0.0/triton-3.0.0-cp310-cp310-win_amd64.whl")
    else:
        # Install PyTorch without CUDA support
        if minor == 11:
            os.system("pip3 install https://download.pytorch.org/whl/cpu/torch-2.2.2%2Bcpu-cp311-cp311-win_amd64.whl#sha256=88e63c916e3275fa30a220ee736423a95573b96072ded85e5c0171fd8f37a755 https://download.pytorch.org/whl/cpu/torchvision-0.17.2%2Bcpu-cp311-cp311-win_amd64.whl#sha256=54ae4b89038065e7393c65bc8ff141d1bf3c2f70f88badc834247666608ba9f4 https://download.pytorch.org/whl/cpu/torchaudio-2.2.2%2Bcpu-cp311-cp311-win_amd64.whl#sha256=6e718df4834f9cef28b7dc1edc9ceabfe477d4dbd5527b51234e96bf91465d9d")
        elif minor == 10:
            os.system("pip3 install https://download.pytorch.org/whl/cpu/torch-2.2.2%2Bcpu-cp310-cp310-win_amd64.whl#sha256=fc29dda2795dd7220d769c5926b1c50ddac9b4827897e30a10467063691cdf54 https://download.pytorch.org/whl/cpu/torchvision-0.17.2%2Bcpu-cp310-cp310-win_amd64.whl#sha256=acad6f9573b9d6b50a5a3942d0145cb0f9100608acb53a09bfc11ed5720dcfe3 https://download.pytorch.org/whl/cpu/torchaudio-2.2.2%2Bcpu-cp310-cp310-win_amd64.whl#sha256=012cd8efbd9e0011abcd79daff98d312136b5e49417062bef1d38cd208f0c05f")
```

**Comment by BBC-Esq** [comment]
Should be resolved in release v5.0.1

**Comment by jafrank88** [comment]
Very cool. I'll try again, but have gotten behind on this project, so it
might be a few weeks.
Jonathan
